### PR TITLE
Login flow

### PR DESCRIPTION
**Pull request by lindajzhao**
Fixes: #88 

- there is login, create account, set password, forget password pages
- remove prev login/logout logic and forms
- save cognito user to the global context

### CODE REVIEW

File: src/App.js [hunk 1]
```diff
@@ -1,50 +1,61 @@
...
+      setAuthState(nextAuthState);
+      setUser(authData);
+    });
+  }, []);
+
   return (
-    <div className="App">
-      <GlobalProvider>
+    <GlobalProvider state={{user, authState}}>
+      <AmplifyContainer>
```
Comment (lindajzhao) [inline src/App.js:37]: AmplifyContainer replaces PrivateRoute/router logic, it requires login automatically

File: src/components/TableRow.js [hunk 2]
```diff
@@ -16,12 +16,10 @@ const ActionButton = (props) => {
...
   if (props.rif) {
     return (
-      <td>
-        {props.children}
-      </td>
+      <td>{props.children}</td>
     );
   } else {
-    return '';
+    return <td></td>;
```
Comment (lindajzhao) [inline src/components/TableRow.js:22]: Fixes React error in console

File: src/constants.js [hunk 3]
```diff
@@ -1,6 +1,13 @@
 
-export default {
+export const constants = {
```
Comment (lindajzhao) [inline src/constants.js:2]: Fixes eslint error

File: src/pages/FileView.js [hunk 4]
```diff
@@ -10,16 +10,18 @@ const FileView = (props) => {
 
   useEffect(() => {
     const fetchData = async () => {
-      const response = await fetch(`${constants.SERVER_DOMAIN}/jobs/${props.id}`, {
+      if (!globalConsumer.token) return;
+
+      const response = await fetch(`${constants.SERVER_DOMAIN}/jobs/${props.match.params.id}`, {
```
Comment (lindajzhao) [inline src/pages/FileView.js:15]: This component wasn't actually reading from url before

**Review by goatandsheep — APPROVED** [review]

**Pull request merged**